crates/astria-conductor/src/executor/mod.rs

```diff
@@ -114,15 +102,15 @@ impl std::fmt::Display for ReaderKind {
 }
 
 impl Executor {
-    pub(crate) async fn run_until_stopped(self) -> eyre::Result<ExitStatus> {
+    pub(crate) async fn run_until_stopped(self) -> eyre::Result<Option<State>> {
```

Comment:
Make a doc comment like this:
```suggestion
/// Run until a stop signal is received or until the stop height is reached.
///
/// The stop height is defined by the execution session that the executor task fetches from its rollup node. If the execution session contains no stop height, executor will run until a stop signal is reached or its execution session gets invalidated by the rollup.
///
/// Returns [`State`] when exiting gracefully (stop signal, stop height reached). `None` is returned if a shutdown signal is received during initialization before a rollup state could be established.
pub(crate) async fn run_until_stopped_or_stop_height_reached(self) -> eyre::Result<Option<State>> {
```

(note the change in the name; it's no longer just `until_stopped`, but we have an actual regular exit case)

Reply:
54d805f